modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition_manager.py

```diff
@@ -51,6 +56,57 @@ def wait_partitions(cls, partitions):
             [block for partition in partitions for block in partition.list_of_blocks]
         )
 
+    @classmethod
+    @_inherit_docstrings(
+        GenericRayDataframePartitionManager.split_pandas_df_into_partitions
+    )
+    def split_pandas_df_into_partitions(
+        cls, df, row_chunksize, col_chunksize, update_bar
+    ):
+        # it was found out, that with the following condition it's more beneficial
+        # to use the distributed splitting, let's break them down:
+        #   1. The distributed splitting is used only when there's more than 6mln elements
+        #   in the `df`, as with fewer data it's better to use the sequential splitting
+        #   2. Only used with numerical data, as with other dtypes, putting the whole big
+        #   dataframe into the storage takes too much time.
+        #   3. The distributed splitting consumes more memory that the sequential one.
+        #   It was estimated that it requires ~2.5x of the dataframe size, so to avoid
+        #   OOM problems, we fall back to sequential implementation in case it doesn't
+        #   fit into memory (using 3x threshold to be on the safe side).
+        distributed_splitting = (
+            (len(df) * len(df.columns)) > 6_000_000
+            and all(is_numeric_dtype(dtype) for dtype in df.dtypes)
+            and psutil.virtual_memory().available > (df.memory_usage().sum() * 3)
+        )
```

Comment:
Reproducing problems can be quite difficult (due to floating dependence on available memory). Maybe add info level logging so we can get more information from the user?

Maybe it's better to increase threshold up to 4? Considering that on Ray at this moment something can be executed in parallel?

Reply:
good point about the logging, added this.

> Maybe it's better to increase threshold up to 4?

I think 4x would be too generous, increased it to 3.5x